### PR TITLE
feat(underglow): RGB toggle controls ext_power

### PR DESCRIPTION
**Pull request by Nicell**
This adds the ability for RGB underglow toggling to control ext_power to save immensely on battery when RGB is off. It also can be turned off if people want to have more fine tuned control if they have more than RGB on their board for example.

This PR also fixes the error being thrown on release of any underglow behavior and the quick blip of red LED on start when using settings.

### CODE REVIEW

**Comment by tominabox1** [comment]
Tested on my QAZ, zero voltage when using the RGB_TOG behavior. Nice work. 

Also testing the power on by default config and that seems to be working as expected also.

File: app/src/rgb_underglow.c [hunk 1]
```diff
@@ -299,6 +299,25 @@ int zmk_rgb_underglow_toggle() {
 
     state.on = !state.on;
 
+#if IS_ENABLED(CONFIG_ZMK_RGB_UNDERGLOW_EXT_POWER)
+    struct device *ext_power = device_get_binding("EXT_POWER");
```
Comment (petejohanson) [inline app/src/rgb_underglow.c:303]: I would just initialize a static variable in the init once, instead of fetching each time.

**Review by petejohanson — CHANGES_REQUESTED** [review]
Just one minor comment. Thanks!

File: app/src/rgb_underglow.c [hunk 2]
```diff
@@ -244,6 +246,11 @@ static int zmk_rgb_underglow_init(struct device *_arg) {
         return -EINVAL;
     }
 
+    ext_power = device_get_binding("EXT_POWER");
```
Comment (petejohanson) [inline app/src/rgb_underglow.c:249]: You should change this to be wrapped in an `#if`, same with the actual declaration of the static variable.

**Review by petejohanson — CHANGES_REQUESTED** [review]
One more item. Thanks!

**Review by petejohanson — APPROVED** [review]
Perfect! Thanks.

**Pull request merged**